java/memory/src/main/java/org/apache/arrow/memory/BufferAllocator.java

```diff
@@ -50,6 +50,27 @@
    */
   public ArrowBuf buffer(int size, BufferManager manager);
 
+  /**
+   * Allocate a new or reused buffer of the provided size. Note that the buffer may technically
+   * be larger or smaller than the  requested size, depending on the rounding option.
+   * However, the buffer's capacity will be set to the configured size.
+   *
+   * @param size The size in bytes.
+   * @param manager A buffer manager to manage reallocation.
+   * @param roundingOption The rounding option.
+   * @return a new ArrowBuf, or null if the request can't be satisfied
+   * @throws OutOfMemoryException if buffer cannot be allocated
+   */
+  public ArrowBuf buffer(int size, BufferManager manager, BaseAllocator.AllocationRoundingOption roundingOption);
+
+  /**
+   * Get the rounded buffer size.
```

Comment:
nit: Gets (https://google.github.io/styleguide/javaguide.html#s7.2-summary-fragment)

Reply:
@emkornfield Thank you for your carefully review. I have revised it accordingly.